### PR TITLE
Block default properties

### PR DESCRIPTION
**Pull request by djih**
1. Update iOS to v4.4.0
2. Update Android to v2.2.0
3. Allow multiple instances
4. Allowing disabling auto-tracking of certain user properties (like city, IP address, etc)

Multiple Instances:
* Tracking a dictionary of Amplitude instances in the Unity Plugin
* Maintains backwards compatibility with `Amplitude.Instance` getter --> maps to `Amplitude.getInstance()` which maps to `Amplitude.getInstance(null)`
* Updated the iOS and Android wrapper code to take in extra instanceName parameter
* Potential for data loss if the `Amplitude.getInstance()` doesn't map correctly to the default instance that already exists in iOS / Android (because then we would have a new set of metadata ie deviceId, userId, etc etc). To guard against this, I keep the instanceName `null` in the Unity Plugin and let the iOS / Android SDKs map the `null` to `$default_instance`. I still use `$default_instance` as the key in the instances dictionary since you can't have a null key value.

Blocking auto tracking user properties
* Each SDK has a helper class to input / track these tracking options
* The Unity SDK API just takes a Dictionary<string, bool> mapping each string option to true to disable. For example `disableCity`: `true` will disable tracking of city. The iOS / Android wrapper code parses this dictionary of string bools and converts them into the respective tracking option helper class. I feel this is probably the most straight forward way to translate the API
* I wrote up some documentation which lists out all the different string keys you can disable

### CODE REVIEW

File: Assets/Amplitude/Amplitude.cs [hunk 1]
```diff
@@ -9,157 +9,183 @@
...
+		if (instances == null) {
+			instances = new Dictionary<string, Amplitude>();
+		}
+
+		Amplitude instance;
+		if (instances.TryGetValue(instanceKey, out instance)) {
+			return instance;
+		}
+		else {
+			instance = new Amplitude(instanceName);
```
Comment (djih) [inline Assets/Amplitude/Amplitude.cs]: This is where I leave `instanceName` as `null` if it's the default instance, but use `$default_instance` as the key in the dictionary

File: Assets/Plugins/iOS/Amplitude/AmplitudeCWrapper.m [hunk 2]
```diff
@@ -47,67 +57,122 @@
...
-        [[Amplitude instance] initializeApiKey:ToNSString(apiKey)];
+        [[Amplitude instanceWithName:ToNSString(instanceName)] initializeApiKey:ToNSString(apiKey)];
+    }
+}
+
+void _Amplitude_setTrackingOptions(const char* instanceName, const char* trackingOptionsJson) {
+    // convert dictionary of tracking options into AMPTrackingOptions object
+    NSDictionary *trackingOptionsDict = ToNSDictionary(trackingOptionsJson);
+    AMPTrackingOptions *trackingOptions = [AMPTrackingOptions options];
+    if (safeGetBoolFromDictionary(trackingOptionsDict, @"disableCarrier", NO)) {
```
Comment (djih) [inline Assets/Plugins/iOS/Amplitude/AmplitudeCWrapper.m]: This translates the Dictionary<string, bool> into an AMPTrackingOptions object

**Comment by djih** [comment]
These are the Android SDK wrapper changes (adding instanceName to all calls, mapping Dictionary<string, bool> into TrackingOptions object): https://github.com/amplitude/Amplitude-Android/pull/176

File: Assets/Amplitude/Amplitude.cs (hunk 1 above)
Comment (curtisliu) [reply Assets/Amplitude/Amplitude.cs]: are we worried about concurrency at all?

**Pull request merged**